### PR TITLE
Generate `jerry.map` file on build

### PR DESCRIPTION
**Pull request by lemmaa**
JerryScript-DCO-1.0-Signed-off-by: Sung-Jae Lee sjlee@mail.com

### CODE REVIEW

**Comment by egavrin** [comment]
What is the reason for this change? Does map generation breaks build on Darwin?

**Comment by lemmaa** [comment]
@egavrin Just added option for convenience in development. Anyway in Darwin, linker option is different to gnu `ld`. `-Map <map-file>` and '-map <map-file>`.

**Comment by galpeter** [comment]
IMHO: we don't need an on/off switch for this. Just generate the map file every time.

File: CMakeLists.txt [hunk 1]
```diff
@@ -234,6 +235,18 @@ project (Jerry CXX C ASM)
...
+  if("${ENABLE_MAP}" STREQUAL "ON")
+ #  if(("${PLATFORM}" STREQUAL "DARWIN") AND (${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang"))
+    if(("${PLATFORM}" STREQUAL "DARWIN") AND (NOT (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)))
+    set(LINKER_FLAGS_COMMON "${LINKER_FLAGS_COMMON} -Xlinker -map -Xlinker jerry.map")
+   else()
+    if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+     set(LINKER_FLAGS_COMMON "${LINKER_FLAGS_COMMON} -Xlinker -Map -Xlinker jerry.map")
+    endif()
+   endif()
+  endif()
```
Comment (galpeter) [inline CMakeLists.txt]: In this case I would prefer the following:
```
if (...) # darwing
   set(MAP_FILE_FLAGS "-Xlinker -map -Xlinker jerry.map")
else ()
   set(MAP_FILE_FLAGS "-Xlinker -Map -Xlinker jerry.map")
endif()
set(LINKER_FLAGS_COMMON "${LINKER_FLAGS_COMMON} ${MAP_FILE_FLAGS}")
```

**Comment by seanshpark** [comment]
+1 for map generation always

**Comment by lemmaa** [comment]
@galpeter thanks! I have updated PR following your opinion. :)

**Comment by egavrin** [comment]
LGTM

**Comment by galpeter** [comment]
lgtm

**Pull request merged**